sql-jdbc/src/main/java/org/opensearch/jdbc/ResultSetImpl.java

```diff
@@ -98,6 +99,26 @@ public ResultSetImpl(StatementImpl statement, List<? extends ColumnDescriptor> c
 
             List<Row> rows = getRowsFromDataRows(dataRows);
 
+            for (int i = 0; i < columnDescriptors.size(); i ++) {
+                if (schema.getOpenSearchType(i) == OpenSearchType.TIMESTAMP ||
+                    schema.getOpenSearchType(i) == OpenSearchType.DATETIME ||
+                    schema.getOpenSearchType(i) == OpenSearchType.TIME ||
+                    schema.getOpenSearchType(i) == OpenSearchType.DATE) {
+                    int maxLength = 0;
+                    for (Row row : rows) {
+                        Object obj = row.get(i);
+                        if (obj != null) {
+                            int len = obj.toString().length();
```

Comment:
is the precision determined by the length of the string or the number of decimal places?

Reply:
https://docs.oracle.com/javase/8/docs/api/java/sql/ResultSetMetaData.html#getPrecision-int-
> For datetime datatypes, this is the length in characters of the String representation (assuming the maximum allowed precision of the fractional seconds component)